flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java

```diff
@@ -235,33 +252,37 @@ public void visit(IfElseStatementNode ifElseStatementNode) {
                 .properties().condition(ifElseStatementNode.condition()).stepOut();
         BlockStatementNode ifBody = ifElseStatementNode.ifBody();
         for (StatementNode statement : ifBody.statements()) {
+            genCommentNode(statement, thenBranchBuilder);
             statement.accept(this);
             thenBranchBuilder.node(buildNode());
         }
+        genCommentNode(ifBody.closeBraceToken(), thenBranchBuilder);
         endBranch(thenBranchBuilder, ifBody);
 
         Optional<Node> elseBody = ifElseStatementNode.elseBody();
         if (elseBody.isPresent()) {
             Branch.Builder elseBranchBuilder =
                     startBranch(If.IF_ELSE_LABEL, FlowNode.Kind.ELSE, Branch.BranchKind.BLOCK,
                             Branch.Repeatable.ZERO_OR_ONE);
-            List<FlowNode> elseBodyChildNodes = analyzeElseBody(elseBody.get());
+            List<FlowNode> elseBodyChildNodes = analyzeElseBody(elseBody.get(), elseBranchBuilder);
             elseBranchBuilder.nodes(elseBodyChildNodes);
             endBranch(elseBranchBuilder, elseBody.get());
         }
 
         endNode(ifElseStatementNode);
     }
 
-    private List<FlowNode> analyzeElseBody(Node elseBody) {
+    private List<FlowNode> analyzeElseBody(Node elseBody, Branch.Builder branchBuilder) {
         return switch (elseBody.kind()) {
-            case ELSE_BLOCK -> analyzeElseBody(((ElseBlockNode) elseBody).elseBody());
+            case ELSE_BLOCK -> analyzeElseBody(((ElseBlockNode) elseBody).elseBody(), branchBuilder);
             case BLOCK_STATEMENT -> {
                 List<FlowNode> elseNodes = new ArrayList<>();
                 for (StatementNode statement : ((BlockStatementNode) elseBody).statements()) {
+                    genCommentNode(statement, branchBuilder);
```

Comment:
Isn't this covered when visiting the statement node?

Reply:
Since we consider the statements in the statement block, we visit the comment before visiting the statement.